File: FundHubClient/src/app/Data/Models/Project.ts
import {User} from "./User";
import {SocialMedia} from "./SocialMedia";
import {Category} from "./Category";

export interface Project {
  id : string
  title : string
  subtitle :string
  description :string
  category : Category
  currentfund : number
  totalfundrequired : number
  email : string
  images : string[]
  userId : string
  user : User
}

File: FundHubAPI/Data/Models/Project.cs
using System.ComponentModel.DataAnnotations;

namespace FundHubAPI.Data.Models;

public class Project
{
    [Key]
    public Guid Id { get; set; }
    public string title { get; set; }
    public string? subtitle { get; set; }
    public string description { get; set; }
    public Guid CategoryId { get; set; }
    public Category category { get; set; }
    public Guid? UserId { get; set; }
    public User? User { get; set; }
    public int currentfund { get; set; }
    public int totalfundrequired { get; set; }
    public string[] images { get; set; }

}